types/context.go

```diff
@@ -286,16 +297,25 @@ func (c Context) TransientStore(key storetypes.StoreKey) KVStore {
 	return gaskv.NewStore(c.MultiStore().GetKVStore(key), c.GasMeter(), c.transientKVGasConfig)
 }
 
-// CacheContext returns a new Context with the multi-store cached and a new
-// EventManager. The cached context is written to the context when writeCache
-// is called. Note, events are automatically emitted on the parent context's
-// EventManager when the caller executes the write.
+// CacheContext returns a new Context with the multi-store cached, a new
+// EventManager, and a new IndexerBlockEventManager. The IndexerBlockEventManager is
+// initialized with the same block height and time as the parent context. The cached context is written
+// to the context when writeCache is called. Note, events are automatically emitted
+// on the parent context's EventManager when the caller executes the write. Also,
+// the parent context's IndexerBlockEventManager is updated with the cached context's
+// IndexerBlockEventManager.
```

Comment:
nit: You can also re-phrasing the comment to talk about both what the `EventManager` and the `IndexerEventManager` does on write in a single sentence.
```suggestion
// IndexerBlockEventManager when the caller executes the write.
```

Reply:
rephrased this comment